inc/migration.class.php

```diff
@@ -1215,4 +1215,130 @@ private function outputMessageToHtml($msg, $style = null, $area_id = null) {
+      $new_fkey  = getForeignKeyFieldForItemType($new_itemtype);
+
+      // Check prerequisites
+      if (!$DB->tableExists($old_table)) {
+         throw new \RuntimeException(
+            sprintf(
+               'Table "%s" does not exists.',
+               $old_table,
+               $new_table
+            )
+         );
+      }
+      if ($DB->tableExists($new_table)) {
+         throw new \RuntimeException(
+            sprintf(
+               'Table "%s" cannot be renamed as table "%s" already exists.',
+               $old_table,
+               $new_table
+            )
+         );
+      }
+      $fkey_column_iterator = $DB->request(
+         [
+            'SELECT' => 'table_name AS TABLE_NAME',
+            'FROM'   => 'information_schema.columns',
+            'WHERE'  => [
+               'table_schema' => $DB->dbdefault,
+               'table_name'   => ['LIKE', 'glpi_%'],
+               'column_name'  => $old_fkey,
+            ],
+            'ORDER'  => 'TABLE_NAME',
+         ]
+      );
+      $fkey_column_array = iterator_to_array($fkey_column_iterator); // Convert to array to be able to loop twice
+      foreach ($fkey_column_array as $fkey_column) {
+         $fkey_table = $fkey_column['TABLE_NAME'];
+         if ($DB->fieldExists($fkey_table, $new_fkey)) {
+            throw new \RuntimeException(
+               sprintf(
+                  'Field "%s" cannot be renamed in table "%s" as "%s" is field already exists.',
+                  $old_fkey,
+                  $fkey_table,
+                  $new_fkey
+               )
+            );
+         }
+      }
+
+      //1. Rename itemtype table
+      $this->displayMessage(sprintf('Rename "%s" table to "%s"', $old_table, $new_table));
+      $this->renameTable($old_table, $new_table);
+
+      //2. Rename foreign key fields
+      $this->displayMessage(
+         sprintf('Rename "%s" foreign keys to "%s" in all tables', $old_fkey, $new_fkey)
+      );
+      foreach ($fkey_column_array as $fkey_column) {
+         $fkey_table = $fkey_column['TABLE_NAME'];
+
+         if ($fkey_table == $old_table) {
+            // Special case, foreign key is inside renamed table, use new name
+            $fkey_table = $new_table;
+         }
+
+         $this->changeField(
+            $fkey_table,
+            $old_fkey,
+            $new_fkey,
+            'integer' // assume that foreign key always uses integer type
+         );
+      }
+
+      //3. Update "itemtype" values in all tables
+      $this->displayMessage(
+         sprintf('Rename "%s" itemtype to "%s" in all tables', $old_itemtype, $new_itemtype)
+      );
+      $itemtype_column_iterator = $DB->request(
+         [
+            'SELECT' => 'table_name AS TABLE_NAME',
+            'FROM'   => 'information_schema.columns',
+            'WHERE'  => [
+               'table_schema' => $DB->dbdefault,
+               'table_name'   => ['LIKE', 'glpi_%'],
+               'column_name'  => 'itemtype',
```

Comment:
Looking at #5954, I guess if we should handle `itemtype_*` fields too.
```suggestion
               [
                  'OR' => [
                     ['column_name'  => 'itemtype'],
                     ['column_name'  => ['LIKE', 'itemtype_%'],
                  ],
               ],
```